packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -363,14 +312,22 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
     function delegateKgh(
         address validator,
         uint256 tokenId
-    ) external isRegistered(validator) returns (uint128, uint128) {
-        uint128 kroInKgh = KGH_MANAGER.totalKroInKgh(tokenId);
-        uint128 kroShares = _convertToKroShares(validator, kroInKgh);
-        uint128 kghShares = _convertToKghShares(validator);
-        _delegateKgh(validator, tokenId, kroInKgh, kroShares, kghShares);
+    ) external isRegistered(validator) returns (uint128) {
+        // TODO: claim boosted reward and delegate to KRO pool
 
-        emit KghDelegated(validator, msg.sender, tokenId, kroInKgh, kroShares, kghShares);
-        return (kroShares, kghShares);
+        KGH.safeTransferFrom(msg.sender, address(this), tokenId);
```

Comment:
I wonder why KGH transfer and `updateValidatorTree` comes out from `_delegateKgh`

Reply:
I wanted the `_delegateKgh` function to do the things only related to delegateKgh.
Transferring KGH and updating validator tree are not a roles of `_delegateKgh`.